### ISSUE TITLE
Unable to replicate SAC SB results in SB3

### ISSUE THREAD

**Issue by vdesai2014** [DLR-RM/stable-baselines3#1201]
### 🐛 Bug

I am trying to use Stable Baselines to learn a grasping policy for a robot and transfer it to a real robot. I found this project by Baris Yazici (https://github.com/BarisYazici/deep-rl-grasping) that was originally done in the older version of Stable Baselines, the agent is able to learn a useful policy starting at around 50,000 timesteps and pretty quickly converges to a high success rate. The original SB observation was passed in as a float from 0 to 1, and SB3 only accepts input of integer type from 0 to 255 for Cnn Policies so I modified the observation in order to match this requirement. The very minor code changes I made are listed in the "Code Example" below, along with the original SB implementation . When using SB3, however, the agent does not learn even after 500,000 timesteps. 

Are folks familiar with differences in implementation of Cnn Policies in SB vs. SB3 able to help with explaining what may be going on here? For whatever reason, training using SB becomes very slow after 20,000 timesteps (1 frame per second), otherwise I would just stick with SB. SB3 runs much faster, but fails to converge. To be honest I'm pretty surprised that the original SB implementation works, as the observation is passed in as a float from 0 to 1, and gets normalized (/= 255) by the VecNormalize function. It seems my issue has to do with the way the observation is changed to be compatible with SB3 CnnPolicy requirements, but I don't see why the minor changes I am making break the learning process. I should note the environment itself is identical in all my trials & hyper parameters are kept constant. 

### Code example

```python
#Original SB Implementation for observation, returns an array float's with shape (64, 64, 2)

 def _observe(self):
         rgb, depth, _ = self._camera.get_state()
         sensor_pad = np.zeros(self._camera.state_space.shape[:2])
         sensor_pad[0][0] = self._actuator.get_state() #gripper_width 
         obs_stacked = np.dstack((depth, sensor_pad))
         return obs_stacked 

 #Original SB Implementation for obs space

 self.observation_space = gym.spaces.Box(low=0, high=255, shape=(shape[0], shape[1], 2))

 #Original SB code that defines policy & initializes training 

 import stable_baselines as sb
 from stable_baselines.sac.policies import CnnPolicy as sacCnn
 from stable_baselines.common.vec_env import VecNormalize

 model = sb.SAC('CnnPolicy',
 self.env,
 verbose=2,
 gamma=0.99,
 buffer_size=1000000,
 batch_size=64,
 learning_rate=0.0003)

 model.learn(total_timesteps=1000000)
```

This next set of code include changes I made to port this over to SB3. Basically just scaling the values by 255 in order for information to be preserved when it gets typecast to uint8.

```python
#SB3 implementation of the observation function 
 def _observe(self):
            rgb, depth, _ = self._camera.get_state()
            scaledepth = depth * 255
            sensor_pad = np.zeros(self._camera.state_space.shape[:2])
            sensor_pad[0][0] = self._actuator.get_state() * 255
            obs_stacked = np.stack((scaledepth, sensor_pad))
            return obs_stacked

#SB3 implementation of obs space 
self.observation_space = gym.spaces.Box(low=0, high=255,
                                                        shape=(2, shape[0], shape[1]), dtype = np.uint8)

#SB3 code that defines policy & initializes training 
import stable_baselines3 as sb
from stable_baselines3.sac.policies import CnnPolicy as sacCnn
from stable_baselines3.common.vec_env import VecNormalize
            model = sb.SAC('CnnPolicy',
                                self.env,
                                verbose=2,
                                gamma=0.99,
                                buffer_size=1000000,
                                batch_size=64,
                                learning_rate=0.0003, device = 'cpu')

            model.learn(total_timesteps=1000000)
```

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a minimal working example to reproduce the bug
- [X] I have checked my env using the env checker
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Hello,
it sounds like you should open an issue on the original repo: https://github.com/BarisYazici/deep-rl-grasping

Could you maybe simplify the code and show what you are doing with SB2 vs what you are doing with SB3 exactly?
(and which one work)
Currently, you display 5 different observation spaces, could you strip that down to one observation space (the one used that manage to converge + minimal training code) vs minimal training code for SB3 that doesn't converge.
That would make it easier to compare.

In the meantime, make sure to use latest SB3 version.

**Comment by vdesai2014** [comment]
Thanks for quick response, I have slimmed my SB3 code down to one case which I feel is virtually identical to the original SB implementation. I think the issue is arising from the fact that the observation from Pybullet for depth comes as a float from 0 to 1. Original SB is able to take this in but since SB3 requires data type of integer from 0 to 255, I have to scale the observation by multiplying by 255. I don't see why this should make the training stop working in SB3 but this is the only difference in a model that converges and one that doesn't (over multiple attempts). 

I opened the issue here because I am actually not changing anything in the Pybullet grasping environment between SB and SB3.

**Comment by araffin** [comment]
> I think the issue is arising from the fact that the observation from Pybullet for depth comes as a float from 0 to 1. Original SB is able to take this in but since SB3 requires data type of integer from 0 to 255,

what error do you get if you don't rescale?
and yes, as written in our doc, we expect by default images when you use a CNN policy and normalization is done automatically, but, you can deactivate the normalization by passing `normalize_images=False` to the policy keyword argument.

> from stable_baselines3.common.vec_env import VecNormalize

you are importing it but not using it, is that intented?

**Comment by araffin** [comment]
oh i see, i think we need to add an argument to bypass the check.
in the meantime, you can pass a custom feature extractor (cf doc) which would be a clone of NatureCNN but without the check.

**Comment by araffin** [comment]
@vdesai2014 I've pushed a fix in https://github.com/DLR-RM/stable-baselines3/pull/1214
the only difference with SB2 is that it requires channel-first input (you can use wrapper for that) and you must pass `normalize_images=False` to explicitly deactivate normalization.
